lib/location/location.c

```diff
@@ -307,6 +307,31 @@ const char *location_method_str(enum location_method method)
 	}
 }
 
+const struct location_data_details *location_details_get(
+	const struct location_event_data *event_data)
+{
+	const struct location_data_details *details = NULL;
+
+#if defined(CONFIG_LOCATION_DATA_DETAILS)
+	switch (event_data->id) {
+	case LOCATION_EVT_LOCATION:
+		details = &event_data->location.details;
+		break;
+	case LOCATION_EVT_TIMEOUT:
+	case LOCATION_EVT_ERROR:
+		details = &event_data->error.details;
+		break;
+	case LOCATION_EVT_FALLBACK:
+		details = &event_data->fallback.details;
+		break;
+	default:
+		/* TODO: think about LOCATION_EVT_RESULT_UNKNOWN */
```

Comment:
Do we want location details for `LOCATION_EVT_RESULT_UNKNOWN`? In practice this would mean cellular and/or Wi-Fi details:
```
"cellular_ncells_count":4,
"cellular_gci_cells_count":0,
"wifi_ap_cnt":4
```

Reply:
It feels a bit awkward and the user has already got the cellular and/or WiFi details with the `LOCATION_EVT_CLOUD_LOCATION_EXT_REQUEST` event.

Reply:
You've got cells and APs with LOCATION_EVT_CLOUD_LOCATION_EXT_REQUEST but you haven't got `mosh_cmd` or `modem_details` if you are using just cellular and Wi-Fi. If GNSS is there also you will see those in that context but you still need to connect those pieces of information from different appId's.

I just started to experiment what to do with `LOCATION_EVT_RESULT_UNKNOWN`...

Reply:
It's two code lines to make the code send location details for `LOCATION_EVT_RESULT_UNKNOWN` if we are OK that those details are in `location_event_data.location_data_error.location_data_details`.

We can still discuss if we include them.

Reply:
I think we could include them.

Reply:
We've decided to keep the data details for `LOCATION_EVT_RESULT_UNKNOWN`.

> It's two code lines to make the code send location details for `LOCATION_EVT_RESULT_UNKNOWN` if we are OK that those details are in `location_event_data.location_data_error.location_data_details`.

As discussed, I've moved the details into `location_event_data.location_data_unknown.location_data_details`.